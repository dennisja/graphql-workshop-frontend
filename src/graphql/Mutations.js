const REGISTER_USERS_MUTATION = gql`
  mutation RegisterUsers(
    $firstName: String!
    $lastName: String!
    $email: String!
    $password: String!
    $username: String!
  ) {
    createUser(
      username: $username
      firstName: $firstName
      lastName: $lastName
      email: $email
      password: $password
    ) {
      user {
        id
        username
      }
    }
  }
`;

const LOGIN_USER_MUTATION = gql`
  mutation LoginUser($username: String!, $password: String!) {
    login(username: $username, password: $password) {
      token
    }
  }
`;

const CREATE_LINK_MUTATION = gql`
  mutation CreateLink($title: String!, $url: String!) {
    createLink(url: $url, title: $title) {
      id
      url
      title
      postedBy {
        id
        firstName
        lastName
        username
      }
    }
  }
`;