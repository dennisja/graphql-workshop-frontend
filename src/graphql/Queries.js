import gql from "graphql-tag";

const GET_LINKS_QUERY = gql`
  query getLinks {
    links {
      id
      url
      title
    }
  }
`;

const GET_USERS_QUERY = gql`
  query getUsers {
    users {
      id
      firstName
      lastName
      email
      username
    }
  }
`;

export { GET_LINKS_QUERY, GET_USERS_QUERY };
