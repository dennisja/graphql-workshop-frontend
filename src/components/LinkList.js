import React from "react";
import { graphql } from "react-apollo";
import Link from "./Link";
import { GET_LINKS_QUERY } from "../graphql/Queries";

export const LinkList = props => {
  const { links, loading, error } = props.links;

  if (loading) return <div className="loading">Loading....</div>;

  if (error) return <div className="error">Error</div>;

  return <div>{links.map(link => <Link key={link.id} link={link} />)}</div>;
};

export default graphql(GET_LINKS_QUERY, { name: "links" })(LinkList);
