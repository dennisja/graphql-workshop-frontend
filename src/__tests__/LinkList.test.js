import React from "react";
import { mount, shallow } from "enzyme";
import { shallowToJson, toJson } from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import LinkList from "../components/LinkList";
import { GET_LINKS_QUERY } from "../graphql/Queries";

describe("User component", () => {
  const wrapper = shallow(<LinkList />);

  it(" renders properly", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("renders the loading screen", () => {
    const request = { query: GET_LINKS_QUERY };
    const wrapper = mount(
      <MockedProvider mocks={[{ request }]}>
        <LinkList />
      </MockedProvider>
    );

    expect(wrapper.find(".loading")).toHaveLength(1);
    expect(wrapper.find(".loading").text()).toEqual("Loading....");
  });

  it(" renders the error component", async () => {
    const request = { query: GET_LINKS_QUERY };
    const error = "Something went Wrong";
    const wrapper = mount(
      <MockedProvider mocks={[{ request, error }]}>
        <LinkList />
      </MockedProvider>
    );
    
    await new Promise(resolve => setTimeout(resolve));
    wrapper.update();
    expect(wrapper.find("div").text()).toEqual("Error");
  });

  it("passes the corerct props", async () => {
    const request = { query: GET_LINKS_QUERY };
    const result = {
      data: {
        links: [
          {
            id: "1",
            url: "https://google.com",
            title: "Google Search",
            postedBy: {
              id: "1",
              firstName: "dennis",
              lastName: "jjagwe",
              username: "dj"
            }
          },
          {
            id: "2",
            url: "https://whatsapp.com",
            title: "Whatsapp Messenger",
            postedBy: {
              id: "1",
              firstName: "dennis",
              lastName: "jjagwe",
              username: "dj"
            }
          }
        ]
      }
    };

    const wrapper = mount(
      <MockedProvider mocks={[{ request, result }]} removeTypename>
        <LinkList />
      </MockedProvider>
    );
    await new Promise(resolve => setTimeout(resolve));
    wrapper.update();
    expect(wrapper.find(LinkList)).toHaveLength(1);
    // console.log(wrapper.find(LinkList).props())
  });
});
