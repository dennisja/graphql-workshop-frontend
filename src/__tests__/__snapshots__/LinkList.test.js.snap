// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component  renders properly 1`] = `
<Query
  displayName="Apollo(Component)"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "getLinks",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "links",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "url",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "title",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "postedBy",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "firstName",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "lastName",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "username",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 161,
        "start": 0,
      },
    }
  }
  skip={false}
  warnUnhandledError={true}
/>
`;
